lib/internal/Magento/Framework/Data/AbstractSearchResult.php

```diff
@@ -59,9 +59,9 @@
     protected $entityFactory;
 
     /**
-     * @var \Magento\Framework\DB\Select
+     * @var \Magento\Framework\DB\Query
      */
-    protected $select;
```

Comment:
We can't remove protected properties now due to backward compatibility policy. Even if these properties are not used.